src/kubernetes/reflector.rs

```diff
@@ -87,6 +97,9 @@ where
                     // begin arriving, reducing the time durig which the state
                     // has no data.
                     self.resource_version.reset();
+                    if let Some(ref mut delayed_delete) = self.delayed_delete {
+                        delayed_delete.clear();
```

Comment:
Why is this necessary? I ask because it seems like we will leak memory with this, in a sense that those entries would never be deleted, unless the whole state is cleared at some point.

Reply:
`self.state_writer.resync()` one line below clears the state. So it's all in line.
I plan to add more tests for resync and deletion logic, to ensure the delayed deletes don't carry over the state resync.

Upon desync, we have to reconstruct the state from scratch, since k8s can't continue from the resource version that we pass to it. The corresponding state operation is *re*sync - it pretty much just clears the state. We postpone the `refresh` to the `evmap` to keep the state populated with stale data, cause it's still better than having an empty state.

Reply:
I see, I thought `resync` was something else. 

Then, couldn't we retain the data, and the delete queue, and let the subsequent events update the data. The only issue that I see is that if an Object was deleted in the meantime we wouldn't get an event for it so it would kinda be like leaked memory. What do you think?

Reply:
That won't work. Kubernetes state can potentially be completely different, no guarantees that whatever they send after a desync applies as a continuation to the state we currently have. Is essence, desync is a request to start from scratch, and we should do accordingly.

We can mitigate the problem with the state briefly being empty during the desync differently - at the state writer level, if we debounce `refresh`es to the `evmap` after every per-item operation (`add`/`remove`/`delete`) we'll be able to "buffer" the state reconstruction, and commit the `evmap` `purge` plus the whole sequence of additions. K8s will send a lot of `add`s quicky in a row when we start a new watch chain with an empty resource version. This way, from the readers' perspective, the state will never be observed in an empty state.

Reply:
That seems good.